.github/workflows/ros-ci.yaml

```diff
@@ -0,0 +1,64 @@
+name: ros-ci
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
```

Comment:
So no CI until you make a PR? I can see how this would save tons of CI time but it also means certain defects won't get found until the PR is made.

Reply:
This runs on all pull_requests and also on main when there is a push into main.